pkg/k8s/apis/cilium.io/v2/register.go

```diff
@@ -135,44 +135,56 @@ func CreateCustomResourceDefinitions(clientset apiextensionsclient.Interface) er
 		},
 	}
 
-	log.Debug("Creating CiliumNetworkPolicy/v2 CustomResourceDefinition...")
-	clusterCRD, err := clientset.ApiextensionsV1beta1().CustomResourceDefinitions().Create(res)
-	if errors.IsAlreadyExists(err) {
-		clusterCRD, err = clientset.ApiextensionsV1beta1().CustomResourceDefinitions().Get(cnpCRDName, metav1.GetOptions{})
+	return createUpdateCRD(clientset, "v2.CiliumNetworkPolicy", res)
+}
+
+// createUpdateCRD ensures the CRD object is installed into the k8s cluster. It
+// will create or update the CRD and it's validation when needed
+func createUpdateCRD(clientset apiextensionsclient.Interface, CRDName string, crd *apiextensionsv1beta1.CustomResourceDefinition) error {
+	scopedLog := log.WithField("name", CRDName)
+
+	clusterCRD, err := clientset.ApiextensionsV1beta1().CustomResourceDefinitions().Get(crd.ObjectMeta.Name, metav1.GetOptions{})
+	if errors.IsNotFound(err) {
+		scopedLog.Info("Creating CRD (CustomResourceDefinition)...")
+		clusterCRD, err = clientset.ApiextensionsV1beta1().CustomResourceDefinitions().Create(crd)
```

Comment:
Is it safe to error out here or do we need to check for `AlreadyExists` again to avoid a race condition causing an unnecessary error?

Reply:
I think that as soon as we `Create`, `Get` won't return `AlreadyExists` error.

Reply:
I think tgraf means a race between agents on different machines. They all come up at once, they all see Get return an error. They now all try to Create, and only one would succeed. I'll add the check to ensure we don't break.